### ISSUE TITLE
[BUG] JS Malfunction when using **SWUP js**

### ISSUE THREAD

**Issue by cavasinf** [tabler/tabler#986]
**Describe the bug**

I'm using [SWUP js](https://swup.js.org/), 
> Think of swup as being pjax on steroids. The concepts are similar to those used in barba.js or other transition libraries, but swup gives more attention to ease of use and flexibility. 

When "pjaxing" the page, we need to reload every JS plugin init/constructor.
https://swup.js.org/getting-started/reloading-javascript

----------------

ATM, we are using/including the `tabler.js` that pre-init all plugins for us.

This file has no exported function and have a self-invoked function, 
so we need to recall the file entirely.

The problem with the current state, 
when recalling the file, code/events will be duplicated/double instantiate.
> Re-running your scripts without destroying remains of previous run can lead to memory leaks and breaking of the page.

It works for plugin like iMask or autosize.
But it bugged out the menu "extend" event: Hard stuck with double click event so it doesn't open the menu.

--------------

To resolve that problem,
- Inits must be on `dom` parameter, not always on `document`.
- Instantiators should be in function/exported function, that we can import from other files.
- Implement a destruct() function ?

Small example how exported/packaged init function with iMask:
```js
/**
 * @param {HTMLElement} dom
 */
export function init(dom) {
    init_iMask(dom);
}

init(document.body);

/**
 * @param {HTMLElement} dom
 */
export function init_iMask(dom){
    var maskElementList = [].slice.call(dom.querySelectorAll('[data-mask]'));
    maskElementList.map(function (maskEl) {
        return new IMask(maskEl, {
            mask: maskEl.dataset.mask,
            lazy: maskEl.dataset['mask-visible'] === 'true'
        });
    });
}
```

**Device**

 - Browser Firefox 95.0.2 (64 bits)
 - OS: Windows 10
 - Screen size 1920x1080

**To reproduce**

1. Reload dynamically a whole block/div that need JS plugin.
2. Include `tabler.js` to init JS plugin for the new block
3. Errors (menu for example)

**Comment by rjd22** [comment]
I personally think this is really SWUP specific. I recommend you rewrite the `tabler.js` to be SWUP compatible since it's not really a big file.

Since this package is build with vanilla js in mind and we will be unable to fix this for every framework in existence and to come, I'm going to close this issue.

**Comment by cavasinf** [comment]
> I personally think this is really SWUP specific.

Not really, example with **modal** that is implemented from **AJAX** HTML content.
If you have JS component in, there's no way to re-init those.

> I recommend you rewrite the `tabler.js`

That's what I've done to use an alternative/temporary way.
